packages/components/table/src/table-body/styles-helper.ts

```diff
@@ -139,17 +148,30 @@ function useStyles<T extends DefaultRow>(props: Partial<TableBodyProps<T>>) {
   const getColspanRealWidth = (
     columns: TableColumnCtx<T>[],
     colspan: number,
-    index: number
+    colspanIndex: number
   ): number => {
     if (colspan < 1) {
-      return columns[index].realWidth!
+      return columns[colspanIndex].realWidth!
     }
-    const widthArr = columns
-      .map(({ realWidth, width }) => realWidth || width)
-      .slice(index, index + colspan)
-    return Number(
-      widthArr.reduce((acc, width) => Number(acc) + Number(width), -1)
-    )
+
+    const spanMethod = parent?.props.spanMethod
+    if (!isFunction(spanMethod)) {
+      const widthArr = columns
+        .map(({ realWidth, width }) => realWidth || width)
+        .slice(colspanIndex, colspanIndex + colspan)
+      return Number(
+        widthArr.reduce((acc, width) => Number(acc) + Number(width))
+      )
+    }
+
+    let totalWidth = 0
+    for (let i = 0; i < colspan; i++) {
+      const physicalIndex = colspanIndex + i
+      const column = columns[physicalIndex]
+      totalWidth += Number(column.realWidth || column.width)
+    }
+
+    return totalWidth
```

Comment:
Whether `spanMethod` is a function or not shouldn't affect the calculation result, right?